File: unused/Node.py
class Node:
    def __init__(self, node_id: int, age: int, hhsize: int,
                 num_cc_nonhh: int, ethnicity: str,
                 disease_status: str = "S", remaining_days_sick: int = 0,
                 connected_nodes=set()):
        self._node_id = str(node_id)
        self._age = age
        self._hhsize = hhsize
        self._num_cc_nonhh = num_cc_nonhh
        self._ethnicity = ethnicity
        self._disease_status = disease_status
        self._remaining_days_sick = remaining_days_sick
        self._connected_nodes = connected_nodes

    @property
    def node_id(self):
        return self._node_id

    @property
    def age(self):
        return self._age

    @property
    def hhsize(self):
        return self._hhsize

    @property
    def num_cc_nonhh(self):
        return self._num_cc_nonhh

    @property
    def ethnicity(self):
        return self._ethnicity

    @property
    def disease_status(self):
        return self._disease_status

    @disease_status.setter
    def disease_status(self, new_status: str):

        if new_status not in ["S", "I", "R"]:
            raise ValueError("new_status must be either S, I, or R")

        self._disease_status = new_status

    @property
    def remaining_days_sick(self):
        return self._remaining_days_sick

    @remaining_days_sick.setter
    def remaining_days_sick(self, new_value=None):
        self._remaining_days_sick = new_value

    def set_sick(self, duration: int = 10):
        self.disease_status = "I"
        self.remaining_days_sick = duration

    @property
    def connected_nodes(self):
        return self._connected_nodes

    @connected_nodes.setter
    def connected_nodes(self, c):
        self._connected_nodes = c

    @connected_nodes.deleter
    def connected_nodes(self):
        self._connected_nodes = set()

    def add_connection(self, node_ids):
        if isinstance(node_ids, int):
            node_ids = {node_ids}
        if isinstance(node_ids, list):
            node_ids = set(node_ids)
        self.connected_nodes = self.connected_nodes.union(node_ids)

File: unused/Node.R
setClass("Node", 
         slots = c(
           id = "numeric", 
           age = "numeric",
           hhsize = "numeric",
           num_cc = "numeric",
           num_cc_nonhh = "numeric",
           ethnicity = "character",
           disease_status = "character",
           remaining_days_sick = "numeric"
         )
) -> Node

setGeneric("set", function(Node, ...) standardGeneric("set"))
setGeneric("getID", function(.Object) standardGeneric("getID"))
setGeneric("disease_status", function(Node) standardGeneric("disease_status"))
setGeneric("disease_status<-", function(Node, value) standardGeneric("disease_status<-"))
setGeneric("remaining_days_sick", function(Node) standardGeneric("remaining_days_sick"))
setGeneric("remaining_days_sick<-", function(Node, value) standardGeneric("remaining_days_sick<-"))
setGeneric("set_sick", function(Node, remaining_days_sick) standardGeneric("set_sick"))


setMethod("set", "Node", function(Node, ...) {
  arguments <- list(...)
  
  for (a in names(arguments)) {
    attr(Node, a) <- arguments[[a]]
  }
  
  return(Node)
})

setMethod("getID", "Node", function(.Object) .Object@id)

setMethod("diseaseStatus", "Node", function(Node) Node@disease_status)

setMethod("diseaseStatus<-", "Node", function(Node, value) {
  Node@disease_status <- value; Node
})

setMethod("remainingDaysSick", "Node", function(Node) Node@remaining_days_sick)

setMethod("remainingDaysSick<-", "Node", function(Node, value) {
  Node@remaining_days_sick <- value; Node
})

setMethod("setSick", "Node", function(Node, remaining_days_sick) {
  diseaseStatus(Node) <- "I"
  remainingDaysSick(Node) <- remaining_days_sick
  Node
})


test_Node <- Node(id = 1, age = 26, hhsize = 2, num_cc = 5, num_cc_nonhh = 4,
                  ethnicity = "white", disease_status = "S", 
                  remaining_days_sick= NA_integer_)
set(test_Node, hhsize = 0)
set_sick(test_Node, 10)
disease_status(test_Node)
